tests/test_local_executor.py

```diff
@@ -360,8 +382,10 @@ def test_finalize_failed_137(docker_cleanup, job, tmp_work_dir, volume_api):
     # impersonate an admin
     docker.kill(local.container_name(job))
 
+    # slightly strange that this is EXECUTED and not ERROR
     wait_for_state(api, job, ExecutorState.EXECUTED)
 
+    # This does run finalize, because the cancelled bit is not set
```

Comment:
This comment seems a bit contextless in this test case?

Reply:
as with the other conversation, was trying to make explicit why this test (`kill`) is different from the other test (now `terminate`), maybe a reword would make that clearer.

Reply:
```suggestion
    # In contrast to test_running_job_terminated_finalized , finalize would be run
    # if a job is externally killed rather than being terminated
```